unidist/core/backends/mpi/core/worker/task_store.py

```diff
@@ -311,6 +315,7 @@ async def execute():
                     else:
                         ObjectStore.get_instance().put(output_data_ids, output_values)
             # Monitor the task execution
+            # a blocking send is used because completion is necessary for processing to continue
```

Comment:
```suggestion
            # We use a blocking send here because we have to wait for
            # completion of the communication, which is necessary for the pipeline to continue.
```